### PR TITLE
Door hacking overhaul

### PR DESCRIPTION
**Pull request by Rukofamicom**
## About The Pull Request

* All doors now have randomized wires upon creation
* Multitool pulsing now has no effect if the door is not powered, instead of only failing to raise bolts. 
* Cutting wires is now guaranteed to produce a shock if the door is currently powered
* Multitool pulsing any of the three "high voltage" wires will now produce a shock if the door is powered.
* Removes "Unknown" from station blueprints - this was caused by Emitters.

## Why It's Good For The Game

* AA should be harder to obtain than crowbar + wirecutters/multitool. This change does not fully tackle this issue, but does up the requirement to include insulation for 100% safe access. 
* Hacking doors takes way too little time currently considering you only need to learn the wires one time near the start of the round. Now each door must be learned individually, but once learned that knowledge may be reused on as many doors as you can memorize throughout the round.
* Working on high voltage equipment should require insulation to perform 100% safely. The changes to shocks maintain multitools as a safer option, no longer guaranteed to be safe.

## Testing Photographs and Procedure

Tested to ensure:
- [x] Multitool functions fail if the door has no power
- [x] Multitool on incorrect wire sends shock
- [x] Wirecutter on any wire shocks unless power is completely severed
- [x] Doors have randomly colored and arranged wires instead of identical arrangements

<details>
<summary> Screenshots + Videos</summary>

Randomized wires

image

Multitool pulsing random wires and then shocking on high voltage wire

https://user-images.githubusercontent.com/9547572/169645261-dad4f9d0-76b2-4665-9f09-c4ea180d47b1.mp4

Multitool pulsing all wires after killing power having no effect

https://user-images.githubusercontent.com/9547572/169645263-d00e050b-0067-45cd-9015-17cd58893f54.mp4

</details>

## Changelog
:cl:
balance: Door hacking now requires insulation to be performed safely.
tweak: Doors no longer share wiring patterns and must have their wires learned/memorized individually.
/:cl:

### CODE REVIEW

**Comment by Rukofamicom** [comment]
Viva la false wall

**Comment by Penwin0** [comment]
Might be nice if they were randomized by departments instead, also please check if the CE blueprints are funky when used in hand to check wires now

**Comment by ToasterBan** [comment]
engineers should have some way to tell wires, engineers often need to unbolt, unshock or even tide into places to repair places, repair of damaged or tided  airlocks or to deconstruct them in the first place

**Comment by PowerfulBacon** [comment]
I mean, engineers get insulated gloves anyway. Also how do the CE's blueprints work now?

**Comment by Rukofamicom** [comment]
> Might be nice if they were randomized by departments instead

Not wholly feasible without a **lot** of mapping work due to the number of airlocks that are both maintenance and departmental. 

> also please check if the CE blueprints are funky when used in hand to check wires now

Checking to be sure at the moment, will add it to the testing section with a checkbox

> engineers should have some way to tell wires, engineers often need to unbolt, unshock or even tide into places to repair places, repair of damaged or tided airlocks or to deconstruct them in the first place

If you have insuls on + wirecutters in one hand and multitool in the other, it's as simple as cutting until you find a power, crowbar afterwards. Unless you hit the bolt wire and don't have a multitool on hand, there is no substantial delay with proper equipment that necessitates engineers be able to see wires by default. 

Ask for this in the skillchip PR if this doesn't convince you - I don't think it's necessary.

File: code/datums/wires/airlock.dm
Comment (PowerfulBacon) [inline code/datums/wires/airlock.dm]: Didn't this already shock you, since electrified shocks you if the UI is open

Comment (Rukofamicom) [reply code/datums/wires/airlock.dm]: My mistake on that one if so, verifying and fixing along with CE blueprint testing.

**Comment by Rukofamicom** [comment]
CE blueprints now work on randomized wires. 

image

**Comment by AnCopper** [comment]
Have you tested the abductor multitool?

File: code/datums/wires/_wires.dm
Comment (PowerfulBacon) [inline code/datums/wires/_wires.dm:256]: CE blueprints now tell you the wires of syndicate bombs

**Comment by Rukofamicom** [comment]
> Have you tested the abductor multitool?

Explicitly? No.
I don't see a reason to either - if the blueprints work the multitool will work. 

```
	// Same for anyone with an abductor multitool.
	else if(user.is_holding_item_of_type(/obj/item/multitool/abductor))
		reveal_wires = TRUE

	// Station blueprints do that too
	else if(user.is_holding_item_of_type(/obj/item/areaeditor/blueprints))
		reveal_wire = TRUE
```

File: code/datums/wires/_wires.dm
Comment (Rukofamicom) [reply code/datums/wires/_wires.dm:256]: Sounds like a good functionality to cut to me then.

As already discussed, it's kinda moot for them to display this since engineers have the right tools as it is and this isn't a function that greatly assists antagonists either - I don't know any that would choose to openly use the blueprints in hand over just manually checking the wires.

I'll see if there's an easier option to implement, but this means I have also enabled the blueprints to view wires of every foreign wiring I realize.

**Comment by AnCopper** [comment]
The CE blueprints should only work for station appliances, by the way.

**Comment by Rukofamicom** [comment]
> The CE blueprints should only work for station appliances, by the way.

Yes, I'm working on deciding a direction for that at the moment. 

Currently they work on anything without randomized wiring, which includes enemy technology. Making all doors randomized removed the blueprints from working on them, I fixed that... and now we have a conflict.

**Comment by Rukofamicom** [comment]
Went ahead and went with disabling blueprints on airlocks. It's not practical or within scope of the PR to introduce the unique airlock types that would be required to preserve this interaction. 

I see three options, because wiring does not respect or interact with door access:
* Make the blueprints work on all airlocks, including off-station ones that they shouldn't be able to identify (This is currently how they function since all doors use the same wiring no matter what, and they work in-hand on all doors while hacking too even if they're off-station)
* Introduce and then map several unique airlock objects and then replace all instances of either on-station or off-station airlocks with new, unique versions. 
* Make the blueprints no longer work on airlocks at all.

I chose option 3.

**Comment by FriendlyContractor** [comment]
I have very mixed feeling about this change. I can see some upsides to it but overall i believe in current state of the game it would be detrimental to overall player experience. I will list some issues and concerns below

- Doors already can be made quite diffucult to hack by reinforcing wire panels with iron and plasteel plates. While iron is fairly easy to remove it still slows down hacking quite a bit and plasteel one requires multiple steps and insulated gloves making door impossible to hack without proper equipment and takes quite some time. I believe this mechanic is enough and there is no need for making hacking doors harder with it already in place
- AI. AI is already huge pain to deal with for antagonists and this gives AI more power than ever making it nigh impossible to escape once bolted. Obviously this affects more than just antagonists. This is huge AI buff and i do not believe its going to be fun for anyone in current state of the game
- This change is crippling for many antagonists and severely limits loadout options for them. For example traitors using contractor kit and heretics need to acess pretty much entire station and this change makes it extremely painfull to play either role(and they are already very difficult) which requires hacking a lot of doors. This gets exponentially worse if you get a role that lacks maintenance acess

**Comment by Rukofamicom** [comment]
> Doors already can be made quite diffucult to hack by reinforcing wire panels with iron and plasteel plates. While iron is fairly easy to remove it still slows down hacking quite a bit and plasteel one requires multiple steps and insulated gloves making door impossible to hack without proper equipment and takes quite some time. I believe this mechanic is enough and there is no need for making hacking doors harder with it already in place

This PR is about how easy doors are to hack from the start - it takes one test door and then for the remainder of the round you are screwdriver -> pulse/cut -> crowbar away from getting into every door on the station. This takes at most about 2 seconds with hotkeys. There's barely time to respond or react even if you are on the other side before the person is already inside. 

> AI. AI is already huge pain to deal with for antagonists and this gives AI more power than ever making it nigh impossible to escape once bolted. Obviously this affects more than just antagonists. This is huge AI buff and i do not believe its going to be fun for anyone in current state of the game

I do concede that this does make AI bolting stronger, but I don't see it as an issue when the focus of the PR is to slow down individual hacking attempts to a more reasonable level. Even in the case of an AI trying to bolt you down in place, I don't think it's good to be able to just reverse that in two seconds with the bare minimum preparation. 
* There is a wire that cuts off the AI connection, indicated by a light. Find and cut it and the AI now has no power on that door - Cut all wires until you see this light change in quick succession and remember there's more than just gloves that insulate you.
* Try to get actual AA instead of relying so heavily on ghetto AA. Captain, HoP, Captain's spare. A really good AI can still lock you up, but prepping via getting actual AA and not running in a straight, predictable line makes it much harder to bolt a door before you're already past it. 

> This change is crippling for many antagonists and severely limits loadout options for them. For example traitors using contractor kit and heretics need to access pretty much entire station and this change makes it extremely painfull to play either role(and they are already very difficult) which requires hacking a lot of doors. This gets exponentially worse if you get a role that lacks maintenance acesss

Okay this one is just makes me a bit sad, because you're talking about how I've hurt antag efficiency without actually preventing them from doing anything. These issues can be solved by *actually engaging* with the crew instead of being laser-focused on speedrunning objectives.
* No maints access? Ask HoP, offer to fix wires and kill rodents. You don't have to *actually* fix wires and kill rodents but does make for a convincing show of good faith toward asking for other things.
* Took contractor or have a target outside of your department? Great, take some time to get close to your target and engage with them. Come up with a reason to be in their department (department codependence will help with this a lot too as we get there). Don't just rush contract after contract; that's hella boring for the crew and your job is to be the star of the show and keep the round entertaining. 
* As a final point in contrast to my other two: false walls still exist. They are not as all-purpose as current hacking is since you can only do them on normal walls in a reasonable amount of time, but they do make for a hella smooth entry and getaway if you weren't otherwise spotted.

**Comment by FriendlyContractor** [comment]
> Try to get actual AA instead of relying so heavily on ghetto AA. Captain, HoP, Captain's spare.
- Stealing all acess is very difficult and not something that can be done reliably. Captain and HoP are both armored, armed and worst of all often difficult to locate and acess. Even if you manage to incapitate them which all and all is not that difficult once you get acess to them you somehow have to do it in a way which will grant you enough time to grab their id and run away without getting caught. Captains spare(on most maps) is locked in a safe behind at least two pairs of plasteel reinforced airlocks in public area which makes it practicaly impossible to break into without attracting attention. You most often simply do not have enough time to take on very difficult task of getting AA and also completing other objectives

> A really good AI can still lock you up, but prepping via getting actual AA and not running in a straight, predictable line makes it much harder to bolt a door before you're already past it.

- First of all you can often get spotted and locked up in middle of commiting a crime which gives you no chance trying to avoid being bolted. Cutting the camera is not an option because it alerts the AI which guarantees its attention and also makes your intentions obvious if crime you are about to commit involves violence. I completly disagree with you saying that being bolted can be avoid by unpredictable movement and even if you somehow manage to do it couple of times you only have to fail once for it to be your doom. To be honest i personally find AI so opressive i will bolt doors open even if i have acess somewhere

> Okay this one is just makes me a bit sad, because you're talking about how I've hurt antag efficiency without actually preventing them from doing anything. These issues can be solved by actually engaging with the crew instead of being laser-focused on speedrunning objectives.

- First of all antagonists i mentioned do not have enough time to engage/roleplay with crew members and complete their objectives at the same time. 90 minutes (assuming round even lasts that long) is often not enough to collect bunch of pierced realities, sacrifice a couple of crewmates and gather materials for rituals all without getting caught. Same thing with contractor kit. You absolutely do not have time to converse with other crew members and much less try to get their trust and gain something this way. I find it to be a good thing where antagonists are very active and challenge the crew making round interesting rather than spend entire round doing nothing or silently complete their objectives often not antagonising other players what so ever. Without antagonists causing chaos all you get to do is same repetetive tasks every round. I think i went perhaps a little off-topic - oh well lets get to the next point.
- You assume other players actually are actually willing to intract with other crew members and game encourages that

> No maints access? Ask HoP, offer to fix wires and kill rodents. You don't have to actually fix wires and kill rodents but does make for a convincing show of good faith toward asking for other things.

- First of all there needs to be a hop for this to work. Second they are under no obligation to grant you the acess( even if they usually do). What if you do not get maintenance acess this way - is your round now screwed ? I dare to say maintenance acess is absolutely essential for antagonists for many purposes and having to spend lots of time hacking each and every maintenance door is absolutely crippling. Like entire maintenance exists because of antagonists and are we gonna make it inacessible to them ?

> Took contractor or have a target outside of your department? Great, take some time to get close to your target and engage with them. Come up with a reason to be in their department (department codependence will help with this a lot too as we get there)

I already made myself clear in case of contractor. Now say you have only one target and are willing to spend overwhelming majority of the round trying to get their trust. In most cases there is absolutely no reason for others to engage with you because they can do everything themselves and you can not come up with false excuse because other players may very well be more knowledgable in your field than you are despite playing different role and your true intentions will be immediately obvious to them.

> Don't just rush contract after contract; that's hella boring for the crew and your job is to be the star of the show and keep the round entertaining.

- a little off topic again but i want to talk about it. In my opinion its quite the opposite - contractor is star of the show if they are very active antagonising the crew and good at it. Nothing fun or exciting about contractor that spends most of the time doing boring and mudane things normal crew members can do

**Comment by Coding-Grape** [comment]
See the problem I have is as you said: false walls
Why would you hack through a door of you can just go around it in just as much time if it's not rwalled or even just deconstruct it.

I feel like this just shifts the meta towards RCD, bombs, and thermite and making engineers lives difficult by having to remember every doors access levels because people are just as likely to destroy the wall or door as hack it

A better choice would indeed not being a lazy person and porting departmental door wires, even just making the doors pull a wire legend from the areas "department", and just make doors outside the areas or in new areas just default to service, maints, hallway legend

**Comment by AnCopper** [comment]
I think one of the underlying issues is that normal walls are simply too weak, why does it require less effort to bust down a god damn wall than it is to hack a door?

**Comment by Rukofamicom** [comment]
> I feel like this just shifts the meta towards RCD, bombs, and thermite and making engineers lives difficult by having to remember every doors access levels because people are just as likely to destroy the wall or door as hack it

RCD is already a massive problem item without this PR that needs to be addressed separately. This PR doesn't make it any worse than it already is, it just makes it more preferable to use *because* it is a problem item. If engis find that even with their safety equipment that ten to fifteen seconds is so unbearable they'd rather destroy their way inside, that's either a player problem or indicative of another problem

taking ten to fifteen seconds to **break into an area you are not supposed to have direct and easy access to** is not unreasonable and nobody is going to change my mind on this. There is also nobody who is going to change my mind that the fact the RCD bypasses access is its own issue. The RCD existing completely trivializes both building of barricades and access, but is a separate problem entirely. 

False walls already take some time to do and only work on a portion of station walls, not all of them. People on the other side of a false wall have time to react to someone breaking in through a wall and can even stop them from making the false wall.

**Comment by karmaisblackandbluepilled** [comment]
This is vile.

File: code/datums/wires/_wires.dm
Comment (Archanial) [inline code/datums/wires/_wires.dm]: commented code + no real reason to remove this

File: code/datums/wires/airlock.dm
Comment (Archanial) [inline code/datums/wires/airlock.dm:100]: this feels like big fuck you to the players
not to mention doubling the damage is 🤨

**Review by Archanial — DISMISSED** [review]
I'd prefer every depart having their own wires instead of every airlock having random wires. (this is suggestion not request)

File: code/datums/wires/airlock.dm
Comment (Rukofamicom) [reply code/datums/wires/airlock.dm:100]: Damage isn't doubled: 50 was an RNG chance, I made it guaranteed.

**Review by ike709 — APPROVED** [review]

**Review by ike709 — DISMISSED** [review]
Time to start my villain arc.

**Comment by PerishedFraud** [comment]
I like this except for the added shocks.

Anti tiding? Very cool.

More noob traps? Eat pants lol. Stupid af.

**Comment by Mat05usz** [comment]
Oh yeah, that's what engineering needed. Massive EMP taking a lot of doors? Now you gotta spend 100x more time fixing doors than before (aren't you happy?). I wonder how is this gonna affect overall gameplay when it comes to fixing areas... Has anyone thought about that?

Edit:
With that funny 'anti tide' PR, you're simply removing any will from any engineer to fix fucking anything. Even maxcaps were basically 'we're not doing shit cause it's too big', now you're making it worse. Good luck finding people to fix your airlocks.

**Comment by DerpiestBruh** [comment]
Assistants when I cover the brig entrance wires with plasteel :)

**Comment by ItsTisme** [comment]
I really dont like that CE blueprints cannot identify wires for airlocks. I'd simply RCD doors and replace them if any thing is wrong with multiple different doors because. Also abductor multitool would become close to useless if they use same logic.

I do think doors need to be harder to hack but this feels a bit extreme and would make engineers job quiet frustrating.

**Comment by Rukofamicom** [comment]
re-implemented the blueprints working on door wires, which includes external door wires.

**Review by ike709 — APPROVED** [review]

**Comment by ike709** [comment]
Need @PowerfulBacon and @Archanial to sign off to unblock merging.

**Comment by FortSumter** [comment]
Booo bad Pr don't merge.

Imo it's completely within reason that NT wouldn't change the wires for every door and wouldn't be worried about it.

From a gameplay standpoint, sometimes you need to be able to hack open a door open when you aren't playing engi. This would turn any room you get trapped in into a perma jail if the AI decides it doesn't want to help you or can't see you. As a professional graytider, I'd also be permanently kept out of my hereditary homeland of maints.

This would also give security even less to do if you made tresspass impossible. I want to arrest people damnit!

**Review by PowerfulBacon — APPROVED** [review]

**Comment by FortSumter** [comment]
Genuine mistake adding this PR

**Comment by ike709** [comment]
Damn that's crazy

**Comment by FortSumter** [comment]
Well guess I've got to learn byond shitcode to make a Pr that nullifies this garbage Pr now.

**Comment by Rukofamicom** [comment]
@FortSumter more to it than making the PR - you also have to come up with a reason that it's good for the game and convince others the reasoning is true. 

* AA should be harder to obtain than crowbar + wirecutters/multitool. This change does not fully tackle this issue, but does up the requirement to include insulation for 100% safe access.
* Hacking doors takes way too little time currently considering you only need to learn the wires one time near the start of the round. Now each door must be learned individually, but once learned that knowledge may be reused on as many doors as you can memorize throughout the round.
* Working on high voltage equipment should require insulation to perform 100% safely. The changes to shocks maintain multitools as a safer option, no longer guaranteed to be safe.

Which points do you disagree with and why?

How is actually taking time and a higher bar of prep to open doors you don't and shouldn't have easy access through is bad for the game?

**Comment by Rukofamicom** [comment]
The next issues that need to be tackled are how deconstructing normal walls and any windows (even before this PR) is such a fast method of entry that doors could be entirely ignored in favor of this.

**Comment by FortSumter** [comment]
@Rukofamicom I do see where you're coming from with this, I just don't agree on it whatsoever. 

The big problem this causes is a balancing issue as there are times when it is actually necessary to force open doors outside of being an engineer. Perhaps there is a plasma flood caused by a malf AI and the only safe place is maints. The AI isn't going to let open the doors for you and all you have are the tools you could print from a lathe, if that. Or perhaps you are an antagonist, and need illicit entry into somewhere but don't want to buy an emag. You get shocked twice every time you mess up a door (which is a bug by the way), and have to wait a few minutes before trying the door again. You could also actually BE an engineer and have to work on the doors for some reason, and now you're stuck in the same situation as everyone else having to work out the wires for Every. Single. Door.

It also doesn't make sense IC because locks in real life are often not as secure as they should be, even electronic ones. Companies like NT despite its status as a mega corp would likely use similar wiring for all but the most secure areas (vault, AI sat, command, ect) because their employees aren't (normally, see antagonist) a major security risk. Breaking and entering is also considered a crime IC, which gives security something to do if someone is doing it without good reason or approval. I just _really_ don't like when coders or admins try to solve a security problem. You are server staff, not IC security officers.

A compromise would be as someone else commented and make each department's doors use different wiring, which is a reasonable thing NT would do as it gives a better level of security while also not driving everyone insane from the stations original designers to the crew that actually have to deal with the doors. One idea I thought of that would also help with balancing would be to give the diagnostic HUDs the ability to see the wires like the blueprints do and let the engineers have access to them. Seeing as the skillchips Pr is in development hell right now, this would be a good way to bypass all the aggravation this causes for the people who _SHOULD_ be working on the doors, while also giving people who _SHOULDN'T_ (read: antagonists) to still be able to get into places suggesting they can obtain one from engineering or one of the two the roboticists can buy.

For walls and windows, maybe windows but walls is a skill issue. On most stations the mapmakers tend to overuse reinforced walls, sometimes using them to wall off entire departments or the exterior of the station which is a bummer as it means you can't expand rooms or change the layout of the station as an engineer for your projects. Normal walls already require you to wear welding helmets or else you'll burn your eyes out. I'm not sure what you have in mind to make it harder (besides maybe making it take longer to wrench the wall frame out) but I feel this could also be handled IC by security if they really want to arrest the scientist walking around with a welding mask and a toolbelt.

Most windows are electrified where relevant, but some main hall facing windows for departments aren't and that does cause an issue; not in the sense of how easy it is, but in how quickly you can do it. Personally I'd leave it alone, but I wouldn't be mad if you made it take longer to remove a window without making it more complicated.


tl;dr Pr is bad for balance and security RP, limit it to department instead of all doors and/or add door wire vision to diagnostic HUDs (and add them to the engineers secure lockers) to balance it.

**Comment by FortSumter** [comment]
I guess you were expecting counter points to the ones you made instead of my essay so I'll also do this:

- There are some instances in which forcing access to a particular department or maintenance is necessary as a non-antag. This also doesn't stop antags who want access to these departments for malicous reasons, as they can 'buy' their own form of access that doesn't require hacking the door. This PR only hurts non-antags and effectively removes the feature from the game.
- Most entrances to major departments are a double set of doors, and a single door to all minor and sub-departments. Even if you get lucky and a random person didn't attempt to stop you or call you out _and_ you didn't husk yourself by pulsing the wrong wire, you still need to get past another door without being noticed or killing yourself. worse, you might kill someone else if they attempt to use the door after you failed to hack it and get you banned if you weren't an antag doing this.
- I don't understand the wording for this one; I thought only cutting wires would shock you unless you pulsed a powerline. At any rate, again there are times when normal crewmembers may need to hack a door. There is also a chance that there may not be enough insuls for every member of engineering, and you can't actually rely on cargo to approve what you requested if you tried to buy them. I'd like to see a return of budget insuls as most maps don't have any, or are a random maint drop.

**Comment by Rukofamicom** [comment]
> Perhaps there is a plasma flood caused by a malf AI and the only safe place is maints. The AI isn't going to let open the doors for you and all you have are the tools you could print from a lathe, if that. Or perhaps you are an antagonist, and need illicit entry into somewhere but don't want to buy an emag.

This shows almost willful ignorance on your part because you're acting like it isn't even remotely possible to open doors in this case. Multitool only shocks on two wires, and a grey bull can prevent shocks the same as insuls while being more limited. It's also possible to go *around* the door rather than through it without shock protection at all. 

In the case of a crisis where you are going to die or be captured if you can't open the door in two seconds: I expect the player to be dead/captured. You were gotten and that's all there is to it. There shouldn't **always** be a way out of every situation, especially not via commonly available tools with no scarcity to them at all.

 > It also doesn't make sense IC because locks in real life are often not as secure as they should be, even electronic ones. Companies like NT despite its status as a mega corp would likely use similar wiring for all but the most secure areas (vault, AI sat, command, ect) because their employees aren't (normally, see antagonist) a major security risk. Breaking and entering is also considered a crime IC, which gives security something to do if someone is doing it without good reason or approval. I just really don't like when coders or admins try to solve a security problem. You are server staff, not IC security officers.
 
The locks are still hardly what one could call secure. The tool requirement to "pick the lock" is higher than before and involves a relatively limited resource. So long as you have those limited resources it still takes very little time to open doors. If you don't have those limited resources (insuls or grey bull specifically), you risk taking some damage but can still relatively safely open a door with a multitool and a bit more time than if you had *all* the tools. 
 
Doors are far from impossible to open, it just takes what I would consider to be an appropriate amount of time to do so now. Two seconds is not an appropriate time. 

> A compromise would be as someone else commented and make each department's doors use different wiring, which is a reasonable thing NT would do as it gives a better level of security while also not driving everyone insane from the stations original designers to the crew that actually have to deal with the doors.

I agree with this as a compromise, but ultimately sided against compromising the intention of the PR due to a combination of:
1) How much remapping would have to be done to accomodate this (regardless of which of the two possible ways to implement it was chosen) 
2) Because of how much less secure it would be. The impact of implementing this change would be borderline negligible and substantially worse than completely random wires. You now have two wires to memorize instead of one in most cases: Maints + target department. Worse, this slows down people who need actual emergency access while not actually slowing down the person who's doing criminal activity. The intention was to slow **both** cases down. 

Breaking in or out is something you should do when you're not being watched (or actively in combat), and something you should have to try and conceal doing. 

> One idea I thought of that would also help with balancing would be to give the diagnostic HUDs the ability to see the wires like the blueprints do and let the engineers have access to them. Seeing as the skillchips Pr is in development hell right now, this would be a good way to bypass all the aggravation this causes for the people who SHOULD be working on the doors, while also giving people who SHOULDN'T (read: antagonists) to still be able to get into places suggesting they can obtain one from engineering or one of the two the roboticists can buy.

Skillchips PR is actually closed now, and I don't think it's going to come back. There's too much resistance to the idea from the powers that be. I'm trying to achieve the same end goal as skillchips through different PRs instead - this is one of them, but it was missing a way for engineers to directly bypass it. I settled on the fact they had the appropriate tools to do speed-hacks as good enough. 

*However*

This idea for an Engi-HUD that can see wires on doors is pretty fantastic and I might go ahead and make that myself. I was trying to play with some way to reveal door wires for engineers that wasn't a skillchip, and adding it as a setting to the multipurpose engi-goggles would cover that pretty well I think. 

> Most windows are electrified where relevant, but some main hall facing windows for departments aren't and that does cause an issue; not in the sense of how easy it is, but in how quickly you can do it. Personally I'd leave it alone, but I wouldn't be mad if you made it take longer to remove a window without making it more complicated.

It's not off the table for doing and I do wish that at least reinforced windows had some extra steps to removing (though not as bad as rwalls), but I don't see going around the door as too much of a problem though because this is intentionally possible in numerous areas. Entering through a normal wall in particular is obnoxiously loud and also takes enough time that people on the other side can react to it (Also false walls are just cool). Entering through a window takes the least amount of time and prep so long as they aren't shocked - time will tell whether this is an issue or not.

**Comment by FortSumter** [comment]
> This shows almost willful ignorance on your part because you're acting like it isn't even remotely possible to open doors in this case. Multitool only shocks on two wires, and a grey bull can prevent shocks the same as insuls while being more limited. It's also possible to go around the door rather than through it without shock protection at all.

From what I remember you get Grey Bulls from hacking a vending machine. You risk killing yourself to get something that can only very briefly keep you from killing yourself. It is also impractical and illogical to take down the _wall_ when there is a perfectly good _door_ you can't open now with this PR and without potentially _killing yourself_. Yes, on the off chance you get lucky and pulse the test wire one the door you can get through and survive whatever mess you're in, but now that's RNG you can die through random chance unless you pre-tested (and statistically died several times from burn damage) every single door on the station and wrote down all of their wires prior to things getting heated.

> In the case of a crisis where you are going to die or be captured if you can't open the door in two seconds: I expect the player to be dead/captured. You were gotten and that's all there is to it. There shouldn't always be a way out of every situation, especially not via commonly available tools with no scarcity to them at all.

It takes most people more than two seconds to hack a door even if they know the wires, and you can't exactly do this if you're being chased by security or an antag. You would have enough time to do this however if the AI bolted you in a room or if there was a plasma flood that hadn't reached you yet. A main design philosophy I constantly encounter in this game is "anything is possible with enough time and determination." If the time is the part that bothers you then make it take longer, not make it impossible to reasonably do.

> The locks are still hardly what one could call secure. The tool requirement to "pick the lock" is higher than before and involves a relatively limited resource.

That source is insulated gloves. Gloves that aren't for public use and security often considers contraband. If you want gloves you either break into engineering or buy them from cargo. Now tell me, would any rational player give someone insuls even if they paid you for them? Unlikely.

> I agree with this as a compromise, but ultimately sided against compromising the intention of the PR

That's sad that it would have to be done that way, but at least that gives my idea as the best alternative which you seem to be on board for. I specifically said the diagnostic HUD as it only has a niche use that shows you when a door is shocked or where a teleport pad is targeted at. I thought instead of making an entirely new item it would be easier to just add the functionality to one that already exists and would make sense while also being an uncommon/rare item. It would mean giving the roboticists the keys to the kingdom, so to speak but they could essentially do that anyway the Mk I Ripley and the drill attachment.

> I don't see going around the door as too much of a problem though because this is intentionally possible in numerous areas.

I can't think of many instances I'd rather remove a wall over hacking open a door. To bypass a maint door like this it usually takes going through 2 walls and unintentionally breaking into a adjacent room and that is only one of several doors to explore maints. Some departments also have their equipment adjacent to the walls and if your intent wasn't to get illicit access to that particular machine then you have a piece of important equipment to take apart just to get into the room.

The only thing I can ask then is can you leave specifically the external airlocks and maints wiring uniform? Security gets a huge buff if certain antags can't get access to places to hide and Nuke Ops need to be able to covertly enter the station without compromising an outer wall. I also like playing assistant and it feels bad to be completely exiled from my homeland although I don't think you care about that.

**Comment by Rukofamicom** [comment]
> From what I remember you get Grey Bulls from hacking a vending machine. You risk killing yourself to get something that can only very briefly keep you from killing yourself. 

This is incorrect. They do not require hacking to access. One sip gives you protection for about 50 seconds. 

> It is also impractical and illogical to take down the wall when there is a perfectly good door you can't open now with this PR and without potentially killing yourself.

I disagree, unless we're talking about an rwall. 

> Yes, on the off chance you get lucky and pulse the test wire one the door you can get through and survive whatever mess you're in, but now that's RNG you can die through random chance unless you pre-tested (and statistically died several times from burn damage) every single door on the station and wrote down all of their wires prior to things getting heated.

There are two wires which shock the user when pulsed. There are four wires that depower the door when pulsed, allowing it to be opened. You have only a 33% chance of being shocked before you find the right wire, and *most* of the time that one shock is not going to be lethal unless the grid is extremely overcharged with power. This is true even if you're so unlucky you get shocked twice before you find one of the four wires. 

> It takes most people more than two seconds to hack a door even if they know the wires, and you can't exactly do this if you're being chased by security or an antag.

It really doesn't. This is how long it takes to completely disable a door for the sake of deconstruction with the right tools, without knowing which wire does what.

https://user-images.githubusercontent.com/9547572/175129581-9934f10e-e7a5-445c-94ff-db7cfaf9c3f6.mp4

12 seconds of clicking and door may be deconstructed:
Cut wires until bolts fall. 
Mend wires until power is back on because I didn't pay attention to when it went off.
Pulse bolts to open
Re-cut all wires except bolts

I am not a practiced greytider either, it just really doesn't take much time at all to do. When people know the wires it takes Screwdriver, Snip/pulse, Crowbar and you're in. That really is only about two seconds unless it was a high security door. 

> That source is insulated gloves. Gloves that aren't for public use and security often considers contraband. If you want gloves you either break into engineering or buy them from cargo. Now tell me, would any rational player give someone insuls even if they paid you for them? Unlikely.

Yes that is kind of my entire point - Pseudo AA shouldn't be extremely easy to come by without some prep. No, people shouldn't just hand you gloves but sometimes they will, especially if you have something to offer. I don't know of many cargo techs that would refuse to exchange some insuls for picking up bounties. 

And yes, sec should confiscate insuls - you don't need to be hacking and you got caught. That's entirely deserved and expected. Consider dying them in a washer so they aren't as obviously insuls. Stash them in a plant if you know you're on arrest and pick them back up afterwards. 

Also again, Grey bull. 

> The only thing I can ask then is can you leave specifically the external airlocks and maints wiring uniform?

External air locks I would be fine with, but not maints. Maints being uniform is explicitly why setting wires by area would do almost nothing - nearly every department has a maints door and would be possible to breach almost anywhere with just one set of wires memorized, although your options are more restricted a bit more to specific doors. 

You're free to make a PR that does this and doing it for specifically external airlocks wouldn't even be that difficult - just a few lines:

```
/datum/wires/airlock/external
	randomize = FALSE
```

and 

```
/obj/machinery/door/airlock/external/Initialize(mapload)
	. = ..()
	wires = new /datum/wires/airlock/external(src)
```

**Comment by FortSumter** [comment]
Sigh... well I guess I can't convince you. Another thought I had was just... making it take longer to screw open the maint panel for the door. Looking forward to the insuls crate getting restricted and grey bull being removed in the next PR. "It's not realistic that a drink can make you immune to being shocked! Only engineers should have access to insulated gloves! Guh!"

But seriously thank you for taking time to address my concerns even if you still decided to keep this PR merged. I really didn't expect you would put in the effort instead of just memeing like certain headmin do on the forum. Also thank you for taking my input on the diagnotic HUD wire display idea as I do think it would be good for balancing this change.

**Comment by Rukofamicom** [comment]
> Another thought I had was just... making it take longer to screw open the maint panel for the door.

This was a suggestion someone else made too, and it definitely would achieve the same end goal, but I personally think it's far less interactive and engaging than actually solving a wire "puzzle" (? can't think of a better word) than simply waiting out some do_after timers. 

> Looking forward to the insuls crate getting restricted and grey bull being removed in the next PR. "It's not realistic that a drink can make you immune to being shocked! Only engineers should have access to insulated gloves! Guh!"

Won't be from me and I'll be pretty disappointed if this does happen, especially anytime soon without similar tools being implemented. Balancing realistic with "is it fun" should be something everyone should ponder when making any change, and while we may disagree in the end it's something I do as well.

**Comment by PerishedFraud** [comment]
lol.

**Pull request merged**
